src/main/java/bio/terra/service/filedata/flight/ingest/IngestCreateBucketForScratchFileStep.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.flight.ingest.SkippableStep;
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.exception.BucketLockException;
+import bio.terra.service.resourcemanagement.exception.GoogleResourceNamingException;
+import bio.terra.service.resourcemanagement.google.GoogleBucketResource;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import java.util.function.Predicate;
+
+public class IngestCreateBucketForScratchFileStep extends SkippableStep {
+
+  private final ResourceService resourceService;
+  private final Dataset dataset;
+
+  public IngestCreateBucketForScratchFileStep(
+      ResourceService resourceService, Dataset dataset, Predicate<FlightContext> skipCondition) {
+    super(skipCondition);
+    this.resourceService = resourceService;
+    this.dataset = dataset;
+  }
+
+  public IngestCreateBucketForScratchFileStep(ResourceService resourceService, Dataset dataset) {
+    this.resourceService = resourceService;
+    this.dataset = dataset;
+  }
+
+  @Override
+  public StepResult doSkippableStep(FlightContext context) throws InterruptedException {
+    FlightMap workingMap = context.getWorkingMap();
+    GoogleProjectResource googleProjectResource =
+        workingMap.get(FileMapKeys.PROJECT_RESOURCE, GoogleProjectResource.class);
+
+    try {
+      GoogleBucketResource bucketForFile =
+          resourceService.getOrCreateBucketForIngestScratchFile(
+              dataset, googleProjectResource, context.getFlightId());
+      workingMap.put(FileMapKeys.INGEST_FILE_BUCKET_INFO, bucketForFile);
+    } catch (BucketLockException blEx) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, blEx);
+    } catch (GoogleResourceNamingException ex) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, ex);
```

Comment:
Not sure if this has been discussed, but my preferred convention is to use `e` for any caught exception (unless you have nested `try`s). I find `e` easier to read, and as a reader I don't get any useful information from `blEx` vs `e`.
```suggestion
    } catch (BucketLockException e) {
      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
    } catch (GoogleResourceNamingException e) {
      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
```